runtime/parachains/src/paras/mod.rs

```diff
@@ -2136,32 +2136,29 @@ impl<T: Config> ParachainsCache<T> {
 		Self { parachains: None, _config: PhantomData }
 	}
 
-	fn ensure_initialized(&mut self) -> &mut Vec<ParaId> {
-		self.parachains.get_or_insert_with(|| Parachains::<T>::get())
+	fn ensure_initialized(&mut self) -> &mut BTreeSet<ParaId> {
+		self.parachains
+			.get_or_insert_with(|| Parachains::<T>::get().iter().cloned().collect())
```

Comment:
Is the cloned necessary? Or could we also do an into_iter() instead of iter()?

Reply:
I don't think the `cloned()` call is necessary at all. Good catch. `into_iter()` seems to work just fine.